packages/flutter_tools/test/general.shard/mdns_discovery_test.dart

```diff
@@ -577,6 +578,27 @@ void main() {
         );
       });
 
+      test('On macOS, throw tool exit with a helpful message when client throws a SocketException on lookup', () async {
+        final MDnsClient client = FakeMDnsClient(
+          <PtrResourceRecord>[], <String, List<SrvResourceRecord>>{},
+          socketExceptionOnStart: true);
+
+        final MDnsVmServiceDiscovery portDiscovery = MDnsVmServiceDiscovery(
+          mdnsClient: client,
+          logger: BufferLogger.test(),
+          flutterUsage: TestUsage(),
+          analytics: const NoOpAnalytics(),
+        );
+
+        expect(
+          portDiscovery.firstMatchingVmService(client),
+          throwsToolExit(message: 'You might be having a permissions issue with your IDE. '
+            'Please try going to '
+            'System Settings -> Privacy & Security -> Local Network -> '
+            '[Find your IDE] -> Toggle ON, then restart your phone.')
```

Comment:
Hm I'm not sure how you're supposed to format a multi line string argument value... Maybe this?

```suggestion
          throwsToolExit(
            message:
              'You might be having a permissions issue with your IDE. '
              'Please try going to '
              'System Settings -> Privacy & Security -> Local Network -> '
              '[Find your IDE] -> Toggle ON, then restart your phone.',
           ),
```

Reply:
i believe u